src/Vehicle/FTPManager.cc

```diff
@@ -257,11 +382,54 @@ void FTPManager::_burstReadFileWorker(bool firstRequest)
     _sendRequestExpectAck(&request);
 }
 
+void FTPManager::_burstWriteFileWorker(bool firstRequest)
+{
+    qCDebug(FTPManagerLog) << "_burstWriteFileWorker: starting burst at offset:firstRequest:retryCount" << _uploadState.expectedOffset << firstRequest << _uploadState.retryCount;
+
+       MavlinkFTP::Request request{};
+       request.hdr.session = _uploadState.sessionId;
+       request.hdr.opcode  = MavlinkFTP::kCmdWriteFile;
+       request.hdr.offset  = _uploadState.expectedOffset;
+
+       if (firstRequest) {
+           _uploadState.retryCount = 0;
+       } else {
+           // Must used same sequence number as previous request
+           _expectedIncomingSeqNumber -= 2;
+       }
+       _uploadState.file.seek(request.hdr.offset);
+       if (_uploadState.file.atEnd()){
+           request.hdr.opcode = MavlinkFTP::kCmdTerminateSession;
+           _sendRequestExpectAck(&request);
+           _uploadState.file.close();
+           _advanceStateMachine();
+       }
+       if (_uploadState.file.size()-request.hdr.offset< MAVLINK_MSG_FILE_TRANSFER_PROTOCOL_FIELD_PAYLOAD_LEN - 11){
```

Comment:
is 11 a known payload offset number? Might be better to declare in some header as a global const or #define